html/pages/device.inc.php

```diff
@@ -229,6 +229,14 @@
                 </a>
               </li>');
 
+        if (@dbFetchCell("SELECT COUNT(stp_id) FROM stp WHERE device_id = '".$device['device_id']."'") > '0') {
```

Comment:
Well it boils down to:
Does `stp` have more than 1 entry?

If so, the count is more excessive than just a boolean check whether something is there or not.

On 8 January 2016 10:52:39 CET, Vitali Kari <notifications@github.com> wrote:
>> @@ -229,6 +229,14 @@
>>                  </a>
>>                </li>');
>>  
>> +        if (@dbFetchCell("SELECT COUNT(stp_id) FROM stp WHERE
>device_id = '".$device['device_id']."'") > '0') {
>
>this should do not count all the id entries, because of  "WHERE
>device_id =" statement in the query.
>As I know the COUNT function is called after WHERE, but you can correct
>me if I'm wrong (I'm not the SQL guru)
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/librenms/librenms/pull/2690/files#r49172512